libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,329 @@
 
-Beats 6.0 introduces a new test command for testing the configuration file. For
-example:
+[[upgrade-mapping-template]]
+==== Upgrade the {es} mapping template
 
-[source,shell]
+Mapping templates and the default index names are versioned. For example,
+Metricbeat {stack-version} typically creates indices like this:
+
+["source","sh",subs="attributes"]
 ------------------------------------------------------------------------------
-metricbeat test config
+metricbeat-{stack-version}-{localdate}
 ------------------------------------------------------------------------------
 
-The old `-configtest` flag is still available, but deprecated.
+And the corresponding {es} template is named +metricbeat-{stack-version}+.
+
+This means that each version of the Beat creates a new index, and it's
+guaranteed that the correct template for that version is applied. With these
+changes in place, you generally don't have to do anything to upgrade the mapping
+template when you move to a new version. Just load the index template *before*
+ingesting any data into {es}.
 
-===== Reference configuration files
+To load the index template, run the `setup` command with the `--template` option
+specified. For example:
 
-The `<beatname>.full.yml` file, which contains all the non-deprecated
-configuration options is renamed to `<beatname>.reference.yml` starting with
-Beats 6.0. We recommend using this file as a reference only. It's not intended
-to be used in production.
+[source,shell]
+----
+metricbeat setup --template
+----
 
-==== Dashboard upgrades
+If your {beats} configuration use an output other than `elasticsearch`,
+temporarily enable the `elasticsearch` output when you run the `setup` command.
+For example:
 
-We recommend re-importing the Kibana dashboards after the Beats and Kibana
-upgrades are complete. This way, you will take advantage of the new Beats
-dashboards created for the 6.0 release.
+[source,shell]
+--
+metricbeat setup --template -E output.logstash.enabled=false -E 'output.elasticsearch.hosts=["localhost:9200"]'
+--
 
-See <<breaking-changes-import-dashboards>> in the Breaking Changes
-documentation.
+[[non-es-outputs]]
+==== How to use versioned templates when the output is not {es}
 
-NOTE: If you've modified the Beats dashboards manually, reimporting them will
-overwrite your changes. Consider saving your dashboards under new IDs, so
-that the reimport doesn't overwrite them.
+If you've configured {beats} to send events to a different output, such as {ls},
+make sure you use versioned templates and indices. Otherwise, after you
+upgrade, there will be field conflicts. To use versioned index templates and
+indices, use the metadata from {beats} to set the index, and allow {beats} to
+manage the template.
 
-==== Heartbeat package names
+For example, for {ls}, set the {ls} `index` setting as recommended in the
+{logstash-ref}/plugins-inputs-beats.html[{beats} input plugin] documentation and
+allow {beats} to manage the index template:
```

Comment:
Should we add a note on how to allow Beats to manage the template. Guide on how to load the template for it is a few lines up (we could link to it).